### ISSUE TITLE
Leaky Relu Slope not in generated bindings

### ISSUE THREAD

**Issue by tiberiusferreira** [LaurentMazare/tch-rs#132]
Hello, thanks for this awesome lib, I love it!

I could not find a way to define the negative slope of the leaky relu. The bindings appear to not generate the version which takes a slope parameter such as: https://pytorch.org/docs/stable/nn.functional.html#leaky-relu . https://github.com/LaurentMazare/tch-rs/blob/2d94d6b7cf8fc9e179515a021ee18c40a310a712/src/wrappers/tensor_generated.rs#L3835-L3842 

Is this indeed missing or am I not looking in the right place.

**Comment by LaurentMazare** [comment]
Right, it's actually the case that parameters with default values are ignored as there is no equivalent in rust (I would be very happy if rust supported named parameters and default values in the same way e.g. ocaml does). Letting the user specify all the parameters would be a bit of a pain too.
Anyway this explains why the slope doesn't appear here. In this case I would just go around it by writing something like the following:
```
fn leaky_relu(t1: Tensor, slope: f64) {
    t1.max1(t1 * slope)
}

**Comment by tiberiusferreira** [comment]
I see. The web-sys crate ran into the same problem and they instead generate different functions:
https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.AudioProcessingEvent.html#method.init_event_with_bubbles_and_cancelable
```Rust
pub fn init_event(&self, type_: &str)

pub fn init_event_with_bubbles(&self, type_: &str, bubbles: bool)

pub fn init_event_with_bubbles_and_cancelable(
    &self, 
    type_: &str, 
    bubbles: bool, 
    cancelable: bool
)
```
It's not perfect, but works quite well. For the relu example it is not a big deal, but could be an issue for some complex operation.

**Comment by LaurentMazare** [comment]
We actually generate multiple functions for overloaded functions. Currently they are just named `fn`, `fn1`, `fn2` ... This is the case for `max` or `squeeze`. The naming convention of web-sys looks quite nicer so I may try switching to this if it's not too complicated.
This would not solve the default parameters issue though, as generating one function for each subset of default parameters may result in generating quite a lot of functions. I'm also not a big fan of the builder pattern so I don't have much constructive ideas at this point.

**Comment by oovm** [comment]
Why I can't define this?

```
fn leaky_relu(t1: Tensor, slope: f64) {
    t1.max1(t1 * slope)
}
```


```yaml
error[E0599]: no method named `max1` found for reference `&Tensor` in the current scope
error[E0277]: the trait bound `Scalar: From<f32>` is not satisfied
   |
67 |         t1.max1(t1 * slope)
   |                    ^ the trait `From<f32>` is not implemented for `Scalar`
   |
   = help: the following other types implement trait `From<T>`:
             <Scalar as From<f64>>
             <Scalar as From<i64>>
   = note: required for `f32` to implement `Into<Scalar>`
   = note: required for `&tch::Tensor` to implement `Mul<f32>`
```

**Comment by LaurentMazare** [comment]
@oovm Some functions have been renamed since then so you may want to try something like:
```rust
fn leaky_relu(t1: Tensor, slope: f64) {
    t1.fmax(&(&t1 * slope))
}
```
